fdbserver/Status.actor.cpp

```diff
@@ -1122,33 +1122,66 @@ ACTOR static Future<JsonBuilderObject> latencyProbeFetcher(Database cx, JsonBuil
-						messages->push_back(JsonString::makeMessage("consistencycheck_disabled", "Consistency checker is disabled."));
-					}
+ACTOR static Future<Void> consistencyCheckStatusFetcher(Database cx, JsonBuilderArray *messages, std::set<std::string> *incomplete_reasons) {
+	try {
+		state Transaction tr(cx);
+		loop {
+			try {
+				tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+				tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+				tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+				Optional<Value> ccSuspendVal = wait(timeoutError(BUGGIFY ? Never() : tr.get(fdbShouldConsistencyCheckBeSuspended), 5.0));
+				bool ccSuspend = ccSuspendVal.present() ? BinaryReader::fromStringRef<bool>(ccSuspendVal.get(), Unversioned()) : false;
+				if(ccSuspend) {
+					messages->push_back(JsonString::makeMessage("consistencycheck_disabled", "Consistency checker is disabled."));
+				}
+				break;
+			} catch(Error &e) {
+				if(e.code() == error_code_timed_out) {
+					messages->push_back(JsonString::makeMessage("consistencycheck_suspendkey_fetch_timeout",
+						format("Timed out trying to fetch `%s` from the database.", printable(fdbShouldConsistencyCheckBeSuspended).c_str()).c_str()));
 					break;
-				} catch(Error &e) {
-					if(e.code() == error_code_timed_out) {
-						messages->push_back(JsonString::makeMessage("consistencycheck_suspendkey_fetch_timeout",
-							format("Timed out trying to fetch `%s` from the database.", printable(fdbShouldConsistencyCheckBeSuspended).c_str()).c_str()));
+				}
+				wait(tr.onError(e));
+			}
+		}
+	} catch(Error &e) {
+		incomplete_reasons->insert(format("Unable to retrieve consistency check settings (%s).", e.what()));
+	}
+	return Void();
+}
+
+ACTOR static Future<Void> logRangeWarningFetcher(Database cx, JsonBuilderArray *messages, std::set<std::string> *incomplete_reasons) {
+	try {
+		state Transaction tr(cx);
+		loop {
+			try {
+				tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+				tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+				tr.setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+
+				Standalone<RangeResultRef> existingDestUidValues = wait(tr.getRange(KeyRangeRef(destUidLookupPrefix, strinc(destUidLookupPrefix)), CLIENT_KNOBS->TOO_MANY));
+				std::map<Key,Key> existingRanges;
+				for(auto it : existingDestUidValues) {
+					KeyRange range = BinaryReader::fromStringRef<KeyRange>(it.key.removePrefix(destUidLookupPrefix), IncludeVersion());
+					if(existingRanges.count(range.begin) && existingRanges[range.begin] == range.end) {
+						messages->push_back(JsonString::makeMessage("duplicate_mutation_streams", format("Backup and DR are not sharing the same stream of mutations for range `%s` - `%s`.", printable(range.begin).c_str(), printable(range.end).c_str()).c_str()));
 						break;
 					}
-					wait(tr.onError(e));
+					existingRanges[range.begin] = range.end;
 				}
+				break;
+			} catch(Error &e) {
+				if(e.code() == error_code_timed_out) {
```

Comment:
I might be missing it, but it doesn't look like anything has a timeout set on it above.